src/NuGet.Core/NuGet.LibraryModel/LibraryDependency.cs

```diff
@@ -169,5 +170,32 @@ public LibraryDependency Clone()
 
             return new LibraryDependency(clonedLibraryRange, Type, IncludeType, SuppressParent, clonedNoWarn, AutoReferenced, GeneratePathProperty, VersionCentrallyManaged, ReferenceType, Aliases);
         }
+
+        /// <summary>
+        /// Merge the CentralVersion information to the package reference information.
+        /// </summary>
+        public static void ApplyCentralVersionInformation(IList<LibraryDependency> packageReferences, IDictionary<string, CentralPackageVersion> centralPackageVersions)
```

Comment:
I like the separation of concerns better here, apply the versions at CPVM population time vs a package spec level. 

It does feel like it could belong in some utility class instead of LibraryDependency. Take this with a grain of salt though.
I'd happy to engage further if others echo the same concerns.